llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -10203,6 +10203,9 @@ class BoUpSLP::ShuffleCostEstimator : public BaseShuffleAnalysis {
     if ((!Root && allConstant(VL)) || all_of(VL, IsaPred<UndefValue>))
       return TTI::TCC_Free;
     auto *VecTy = getWidenedType(ScalarTy, VL.size());
+    if (auto Cost = TTI.getBuildVectorCost(VecTy, VL, CostKind);
```

Comment:
I'd be happy to hear alternatives to use the existing `getVectorInstrCost` interface better. I had tried pattern matching `InsertElt(InsertElt(poison, a, 0), b, 1)` but found that the second insert is also called with poison, so we can't pattern match it. e.g. here both calls are passed poison:
https://github.com/llvm/llvm-project/blob/6bfedfa0ba31a8ac8fd7fcfd2d33afaa5eabe0e5/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp#L5883-L5888

Reply:
Thanks, that's exactly what I was looking for!